src/tasks/tasks.hpp

```diff
@@ -207,17 +262,30 @@ class TaskList {
     return AddTask(TaskQualifier::normal, dep, std::forward<Args>(args)...);
   }
 
+  template <class Arg1, class... Args>
+  TaskID AddTask(const TaskQualifier tq, TaskID dep, Arg1 &&arg1, Args &&...args) {
+    if constexpr (std::is_invocable<Arg1, Args...>::value) {
+      return AddTaskImpl(tq, dep, {}, std::forward<Arg1>(arg1),
+                         std::forward<Args>(args)...);
+    } else if constexpr (std::is_convertible<Arg1, std::string>::value) {
+      return AddTaskImpl(tq, dep, std::forward<Arg1>(arg1), std::forward<Args>(args)...);
+    } else {
+      PARTHENON_FAIL("Bad signature for AddTask.");
```

Comment:
Can this be
 ```suggestion
      static_assert(false, "Bad signature for AddTask.");
```
or something similar? Since this is `constexpr` it should be possible to make this thing fail at compile time.

You could maybe `static_assert` for the two conditions in the if branches before branching?

Reply:
Yeah, it is possible but you run into the `static_assert(false)` makes compilation fail no matter which branch is taken. Need to do something like `static_assert(alway_false_t<Arg1>::value)`. I will switch it, I was just being lazy when I wrote this.